### ISSUE TITLE
Issue with isoform quantification

### ISSUE THREAD

**Issue by olazaro-ibri** [GoekeLab/bambu#463]
Hello, previously with running bambu I was able to identify novel transcripts no problem (running the bambu command with quant = F) but now I'm needing to find the expression levels for each of the isoforms from my results, so I'm running a revised form of the bambu command. However, it's throwing me an error (or series of errors) at this after I generate the model as follows:

--- Start extending annotations ---
The current filtering criteria filters out all new read 
            classes, please consider less strigent critria!
--- Start isoform quantification ---
Error: BiocParallel errors
  5 remote errors, element index: 1, 2, 3, 4, 5
  0 unevaluated and other errors
  first remote error:
Error in filter(., (uniqueStartLengthQuery <= primarySecondaryDistStartEnd & : In argument: `==...`.
In addition: Warning message:
Returning more (or less) than 1 row per `summarise()` group was deprecated in
dplyr 1.1.0.
ℹ Please use `reframe()` instead.
ℹ When switching from `summarise()` to `reframe()`, remember that `reframe()`
  always returns an ungrouped data frame and adjust accordingly.
ℹ The deprecated feature was likely used in the bambu package.
  Please report the issue to the authors. 
Execution halted

I'm confident that there aren't any errors with the bam files I'm contributing as I was able to get reliable results when only finding the transcripts, so any help would be appreciated

**Comment by cying111** [comment]
Hi @olazaro-ibri ,

Based on the output message, something unexpected might be happening during processing, possibly due to how the arguments are specified.

Could you share the commands you used to run Bambu for the quantification step? If possible, the commands for the discovery step would also be helpful.

**Comment by olazaro-ibri** [comment]
No problem, I'm using 2 lists of bam files (rc_files_ad and rc_files_control) and my code is the following:

gtf_file <- "Homo_sapiens.GRCh38.107_ERCC.gtf"
fa_file <- "GCF_000001405.40_GRCh38.p14_genomic.fna"

bambuAnnotations <- prepareAnnotations(gtf_file)

se_ad <- bambu(reads=rc_files_ad, annotations=bambuAnnotations, genome=fa_file,
            ncore = 8) 
se_c <- bambu(reads=rc_files_control, annotations=bambuAnnotations, genome=fa_file,
              ncore = 8)

**Comment by cying111** [comment]
Hi,

may I clarify with you when running with quant = FALSE, what is the output message like?

**Comment by olazaro-ibri** [comment]
The output message for that is normal, I'm able to write out my results to a GTF file and I'm able to pull isoforms from the output no problem. I can get specifics if you need, but it doesn't throw me any notable warnings besides "Detected 25 warnings across the samples during read class construction. Access warnings with metadata(bambuOutput)$warnings"

**Comment by cying111** [comment]
Hi, 

could you do me a favor by running the command with quant = TRUE and verbose = TRUE, and then share with me the complete output from bambu? Btw, which bambu version are you using? For checking bambu version, you can simply run sessionInfo()

Thanks!

**Comment by olazaro-ibri** [comment]
I'll let you know the results from the command. I am using bambu v3.4.1 in R v4.3.1

**Comment by olazaro-ibri** [comment]
Alright, after running with those arguments, these are the error(s) I'm getting:

Sample is missing presence of both TRUE and FALSE labels.
Sample does not have more than 50 of both read class labels
Transcript model not trained. Using pre-trained models
Finished generating scores for read classes in 0.4 mins.

--- per sample warnings during read class construction ---
Warnings for: s5292_alignment
no annotations with matching seqlevel styles, all missing chromosomes will use de-novo annotations
not all chromosomes present in reference annotations, annotations might be incomplete. Please compare objects on the same reference
43675 reads are mapped outside the provided genomic regions. These reads will be dropped. Check you are using the same genome used for the alignment
No aligned spliced reads detected!Bambu expects spliced reads. If this is intended, see Documentation on how to handle single-exon transcripts
Bambu was unable to train a model on this sample, and is using a pretrained model

I am using the same .fa file I used to align the reads using minimap2

**Comment by cying111** [comment]
Hi,

the output seems to suggest that there is a mismatch between fasta/gtf file and your rds files in terms of seqlevel styles, as indicated by the message quoted below:  

> Sample is missing presence of both TRUE and FALSE labels.
> Sample does not have more than 50 of both read class labels
> no annotations with matching seqlevel styles, all missing chromosomes will use de-novo annotations
> 43675 reads are mapped outside the provided genomic regions. These reads will be dropped.


 I would suggest to check if the seqlevel style is matched between fasta, gtf, and your samples data. After making sure that the seqlevel style is matched, you can re-run bambu to see if error persists.

**Comment by olazaro-ibri** [comment]
This is what worked, thanks so much! I checked the seqlevelsStyle on the gtf/fa files I was using and they were mismatched, so I switched my annotation file to match it and was able to get a quantification output.
